akirachix/Taishi-Frontend
# added signup and login form

# Engineers Checklist


## What this PR does?
This PR adds a new signup and login functionality to the Next.js application. This includes the implementation of the signup and login pages, integration with the backend authentication system (e.g., NextAuth.js, Auth0), and error handling for user authentication. This update aims to enhance the user experience by providing a secure and seamless way for users to create accounts and access the application.

- List of updates for features, enhancements, or bugfixes

1. Implements the user signup and login process.
2. Fixes related navigation issues for authentication pages.
3. Provides validation feedback to users during authentication.

Dependencies
NextAuth.js and Auth0: Used for managing user authentication.
React Hook Form / yup: For handling form state and validation on the signup and login pages.
Fetch API: For communicating with the backend authentication endpoints.


What this PR Does
Adds signup and login pages to the Themis AI application.
Integrates form validation to ensure proper user input.
Connects the frontend to the backend authentication system.
Displays user feedback for errors (e.g., invalid credentials, missing fields).

How Are You Doing This?
Signup Page: Created with form input fields for email, password, firstname and lastname. Utilizes form validation to ensure user input meets requirements before submission.
Login Page: Provides fields for email and password with real-time validation feedback.
Backend Integration: Uses API routes in Next.js to communicate with the authentication provider (NextAuth.js or Auth0) to handle user creation and login.


Please delete options that are not relevant.



## How can it be tested?
Run the Application: Start the Next.js server using npm run dev.
Navigate to Signup Page: Go to /signup and fill in the form fields (email,  password, firstname, lastname). Test for validation messages.
Submit Signup Form: Verify that successful registration redirects the user appropriately or displays success messages.
Navigate to Login Page: Go to /login and enter credentials.
Submit Login Form: Verify the login process by checking session creation and access to protected pages.
Error Handling: Test with invalid inputs to confirm error messages are displayed correctly.

# Where is it documented?
https://lavenderanyango.atlassian.net/browse/TG-141
https://lavenderanyango.atlassian.net/browse/TG-142
